core/src/bls_sigverify/bls_sigverifier.rs

```diff
@@ -90,6 +92,7 @@ pub struct BLSSigVerifier {
     stats: BLSSigVerifierStats,
     verified_certs: RwLock<HashSet<Certificate>>,
     vote_payload_cache: RwLock<HashMap<Vote, Arc<Vec<u8>>>>,
+    consensus_metrics: Arc<PlRwLock<ConsensusMetrics>>,
```

Comment:
Is it very expensive to create a local consensus_metrics here so you don't have to pass it all the way from validator?

Reply:
It has to be the same shared instance.  In `votor` crate, we are recording the start of slot.